### PR TITLE
Homework project week1 costa

### PR DESCRIPTION
**Pull request by magnicel**

### CODE REVIEW

**Review by ObadaMoustafa — COMMENTED** [review]
Hello Costa I tried to open your Project but I got two errors Please fix and push and let me know to be able to review your PR

**Comment by magnicel** [comment]
fixed Homework-week1

File: week1/project/ecommerce/src/App.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+function App() {
+  const [state, setState] = useState({selectCategory:'', isAll: true})
+  return (
+    <div className = "App">
+      <h1>Products</h1>
+      <Categories categories={categories} state={state} setState={setState} />
+      <ProductsList products={products} state={state} />
+    </div>
+  );
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/App.js:16]: very clear. nice 💥

File: week1/project/ecommerce/src/components/Category.jsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+function Category  ({data, className, onClick})  {
+    return (
+        <div className={`category-item ${className}`} onClick={() => onClick(data)}>{data}</div>
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/components/Category.jsx:3]: smart that you made the `category-item` class is fixed

File: week1/project/ecommerce/src/components/Product.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+function Product({image, title, description}) {
+    return (
+      <li className='product-item'>
+        <div className='product'>
+          <img className='product-image' src={image} alt={description} />
+          <span className='product-title' title={title}>{title}</span>
+        </div>
+      </li>
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/components/Product.jsx:8]: great that you did `Product` component for every product

File: week1/project/ecommerce/src/components/Categories.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+                selectCategory: newCategory, 
+                isAll: (prevState.selectCategory === newCategory) && !prevState.isAll
+            }
+        })
+    }
+    
+    return (
+        <div className="categories">
+            {categories.map((item, index) => {
+                const className = ((item === state.selectCategory) && !state.isAll ? 'category-select' : '')
```
Comment (ObadaMoustafa) [inline week1/project/ecommerce/src/components/Categories.jsx:18]: smart logic to inject a class to the selected category

**Review by ObadaMoustafa — COMMENTED** [review]
great job Costa.
it works perfectly and you code is clean and readable

perfect 🚀

**Pull request closed without merging**